tools/mason/MasonExternal.chpl

```diff
@@ -41,14 +47,40 @@ proc masonExternal(args: [] string) {
       exit(0);
     }
     else if args[2] == "--setup" {
-      if isDir(SPACK_ROOT) then
+      // If spack and spack registry is present with latest version, print message
+      if isDir(SPACK_ROOT) &&
+      isDir(MASON_HOME+'/spack-registry') &&
+      getSpackVersion == spackVersion &&
+      SPACK_ROOT == spackDefaultPath {
         throw new owned MasonError("Spack backend is already installed");
-      else if MASON_OFFLINE then
```

Comment:
Is `MASON_OFFLINE` still respected with these changes? Setting `MASON_OFFLINE` should disable `mason external --setup` still.

Reply:
Done 👍